### PR TITLE
[PM-9170] combine library migrations

### PR DESCRIPTION
**Pull request by audreyality**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-9170

## 📔 Objective

Combines the following PRs to avoid DI failures due to misalignment between angular libraries and consuming apps:

* #9669
* #9666
* #9662
* #9638
* #9637

**Note:** All of the above PRs have already been reviewed. This PR just smashes them all together on the latest code from `main`, fixes the resulting conflicts, and reverts the factory rename in the CLI.

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Comment by audrey-jensen** [comment]
♻️ Sometimes I need to learn things the hard way.

File: libs/tools/generator/extensions/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export * as history from "./history";
+export * as legacyPassword from "./legacy-password";
+export * as legacyUsername from "./legacy-username";
+export * as navigation from "./navigation";
+export { PasswordGenerationServiceAbstraction } from "./legacy-password";
+export { UsernameGenerationServiceAbstraction } from "./legacy-username";
```
Comment (djsmith85) [inline libs/tools/generator/extensions/src/index.ts:6]: This file needs to be deleted, as we now have individual packages for all extensions which provide their exports

File: libs/tools/generator/extensions/legacy/package.json [hunk 2]
```diff
@@ -19,8 +19,8 @@
   },
   "dependencies": {
     "@bitwarden/common": "file:../../../common",
-    "@bitwarden/generator-core": "file:../core",
-    "@bitwarden/generator-history": "file:../extensions/history",
-    "@bitwarden/generator-navigation": "file:../extensions/navigation"
+    "@bitwarden/generator-core": "file:../../core",
```
Comment (djsmith85) [inline libs/tools/generator/extensions/legacy/package.json:22]: This change also needs to be done on the `history/package.json` and `navigation/package.json`

**Review by djsmith85 — CHANGES_REQUESTED** [review]
Unfortunate that you had to bunch all changes into one PR again. Luckily the majority of changes are just changing imports.

Reviewed all changed files and besides two small things for future package-management, this looks good!

**Review by jprusik — DISMISSED** [review]
LGTM for Autofill concerns

BIT run:
https://github.com/bitwarden/browser-interactions-testing/actions/runs/9663985511

File: libs/tools/generator/extensions/src/index.ts (hunk 1 above)
Comment (audrey-jensen) [reply libs/tools/generator/extensions/src/index.ts:6]: Hmm. I did delete it, but the 🐱 came back, the very next merge...

File: libs/tools/generator/extensions/legacy/package.json (hunk 2 above)
Comment (audrey-jensen) [reply libs/tools/generator/extensions/legacy/package.json:22]: Addressed in [6e91216](https://github.com/bitwarden/clients/pull/9812/commits/6e912161bd441ff3db5583efbf39cd4127bf4e18).

**Review by djsmith85 — APPROVED** [review]

**Review by addisonbeck — APPROVED** [review]

**Review by jprusik — APPROVED** [review]

**Review by gbubemismith — APPROVED** [review]

**Comment by audreyality** [comment]
📝 Merged `main` to fix the macOS build errors.

**Review by rr-bw — APPROVED** [review]

**Pull request merged**